cfn-resources/federated-database-instance/test/cfn-test-create-inputs.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+# Copyright 2023 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# cfn-test-create-inputs-with-clusters.sh
+#
+# This tool generates json files in the inputs/ for `cfn test`.
+#
+
+set -o nounset
+set -o pipefail
+set -x
+
+function usage {
+    echo "usage:$0 <project_name>"
+}
+
+if [ "$#" -ne 1 ]; then usage; fi
+if [[ "$*" == help ]]; then usage; fi
+
+rm -rf inputs
+mkdir inputs
+
+#set profile
+profile="default"
+if [ ${MONGODB_ATLAS_PROFILE+x} ];then
+    echo "profile set to ${MONGODB_ATLAS_PROFILE}"
+    profile=${MONGODB_ATLAS_PROFILE}
+fi
+
+#project_id
+projectName="${1}"
+projectId=$(atlas projects list --output json | jq --arg NAME "${projectName}" -r '.results[] | select(.name==$NAME) | .id')
+if [ -z "$projectId" ]; then
+    projectId=$(atlas projects create "${projectName}" --output=json | jq -r '.id')
+
+    echo -e "Created project \"${projectName}\" with id: ${projectId}\n"
+else
+    echo -e "FOUND project \"${projectName}\" with id: ${projectId}\n"
+fi
+
+keyRegion=$AWS_DEFAULT_REGION
+if [ -z "$keyRegion" ]; then
+keyRegion=$(aws configure get region)
+fi
+
+# shellcheck disable=SC2001
+keyRegionUnderScore=$(echo "$keyRegion" | sed -e "s/-/_/g")
+keyRegionUnderScore=$(echo "$keyRegionUnderScore" | tr '[:lower:]' '[:upper:]')
+echo "$keyRegion"
+
+
+
+echo -e "--------------------------------create aws bucket document starts ----------------------------\n"
+bucketName="cfn-boto-data-federation-test1-${keyRegion}"
+aws s3 rb "s3://${bucketName}" --force
+aws s3 mb "s3://${bucketName}" --output json
+echo -e "--------------------------------create aws bucket document  ends ----------------------------\n"
+
+roleID=$(atlas cloudProviders accessRoles aws create --projectId "${projectId}" --output json | jq -r '.roleId')
+echo -e "--------------------------------Mongo CLI Role creation ends ----------------------------\n"
+
+
+echo -e "--------------------------------create key and key policy document starts ----------------------------\n"
+roleName="cfn-boto-df-role-${keyRegionUnderScore}"
```

Comment:
I think this is the issue - can you try again using a role prefix of 'mongodb-atlas-' - permissions are restrict to a subset of roles for tests